tests/lib/rules/no-unused-vars.js

```diff
@@ -37,21 +37,39 @@ ruleTester.defineRule("use-every-a", context => {
 /**
  * Returns an expected error for defined-but-not-used variables.
  * @param {string} varName The name of the variable
+ * @param {string} [additional] The additional text for the message data
  * @param {string} [type] The node type (defaults to "Identifier")
  * @returns {Object} An expected error object
  */
-function definedError(varName, type) {
-    return { message: `'${varName}' is defined but never used.`, type: type || "Identifier" };
+function definedError(varName, additional, type) {
+    return {
+        messageId: "unusedVar",
+        data: {
+            varName,
+            action: "defined",
+            additional: additional || ""
```

Comment:
Could these default values be default parameter values instead? e.g.
```js
function definedError(varName, additional = "", type = "Identifier") {
```

Reply:
I focused on doing just the `messageId`s as the initial goal of this PR, but am happy to do additional refactoring if desired.

I noticed a few of the rules look like they could use a bit of love, so more than happy to do a follow up PR in the same vein as this one focusing on some general light-spring-cleaning PR if you'd like :)

Reply:
That sounds amazing :+1: